ez-launch/sources/ez_launch.move

```diff
@@ -0,0 +1,278 @@
+        } else {
+            option::none()
+        };
+        let constructor_ref = if (option::is_some(&max_supply)) {
+            collection::create_fixed_collection(
+                creator,
+                description,
+                option::extract(&mut max_supply),
+                name,
+                royalty,
+                uri,
+            )
+        } else {
+            collection::create_unlimited_collection(
+                creator,
+                description,
+                name,
+                royalty,
+                uri,
+            )
+        };
+
+        collection_components::create_refs(
+            &constructor_ref,
+            mutable_collection_metadata, // mutable_description
+            mutable_collection_metadata, // mutable_uri
+            mutable_collection_metadata, // mutable_royalty
+        );
+
+        let collection_properties = collection_components::create_properties(
+            mutable_collection_metadata, // mutable_description
+            mutable_collection_metadata, // mutable_uri
+            mutable_token_metadata, // mutable_token_description
+            mutable_token_metadata, // mutable_token_name
+            mutable_token_metadata, // mutable_token_properties
+            mutable_token_metadata, // mutable_token_uri
+            mutable_collection_metadata, // mutable_royalty
+            true, // tokens_burnable_by_creator, just in case creator makes a mistake in minting?
+            true, // tokens_transferable_by_creator, so that we could pre mint to creator and then transfer to user as user mint
+        );
+        collection_components::init_collection_properties(&constructor_ref, collection_properties);
+
+        let object_signer = object::generate_signer(&constructor_ref);
+
+        move_to(&object_signer, CollectionDetails {
+            to_be_minted_vec: vector[],
+            coin_payments: vector[],
+            random_mint,
+            is_soulbound,
+        });
+
+        let object_addr = signer::address_of(&object_signer);
+        let collection_details = borrow_global_mut<CollectionDetails>(object_addr);
+        
+        if (option::is_some(&mint_fee)) {
+            let mint_fee_category = b"Mint Fee";
+            let coin_payment = coin_payment::create<AptosCoin>(option::extract(&mut mint_fee), creator_addr, string::utf8(mint_fee_category));
+            vector::push_back(&mut collection_details.coin_payments, coin_payment);
+        };
+
+        let length = vector::length(&token_name_vec);
+        let i = 0;
+        let collection = object::object_from_constructor_ref(&constructor_ref);
+        let collection_address = object::object_address<Collection>(&collection);
+
+        move_to(&object_signer, EZLaunchRefs { extend_ref: object::generate_extend_ref(&constructor_ref) });
+
+        // pre mint all the tokens to the collection address
+        while (i < length) {
+            let token = pre_mint_helper(
```

Comment:
agree on pre-mint as we separate entry call to call against, not coupled with collection creation, but dont think it should be designed this way as it leaves out a key piece, where are token metadata passed in? if its passed in by this func, only creators can call, otherwise anyone could dump trash tokens to collections. 

other than that, the proposal makes sense with keeping track of the stages and separate pre_mint call. 

btw it's also challenging to track whether all tokens are pre-minted successfully, as collection could be unlimited, so we should leave to sdk to batch call this func iteratively instead